src/model/OWL.plt

```diff
@@ -0,0 +1,547 @@
+  assert_true(is_individual(test:'bIndiv')),
+  assert_false(is_individual(test:'P')),
+  % Argument is unbound
+  assert_true(is_individual(_)),
+  assert_true(once((is_individual(A), test:'bIndiv'=A))).
+
+test('is_object_property') :-
+  assert_false(is_object_property(test:'E')),
+  assert_false(is_object_property(owl:'Thing')),
+  assert_false(is_object_property(owl:'Nothing')),
+  assert_false(is_object_property(test:'DoesNotExists')),
+  assert_true(is_object_property(test:'p')),
+  assert_false(is_object_property(test:'bIndiv')),
+  assert_false(is_object_property(test:'P')),
+  % Argument is unbound
+  assert_true(is_object_property(_)),
+  assert_true(once((is_object_property(A), test:'p'=A))).
+
+test('is_data_property') :-
+  assert_false(is_data_property(test:'E')),
+  assert_false(is_data_property(owl:'Thing')),
+  assert_false(is_data_property(owl:'Nothing')),
+  assert_false(is_data_property(test:'DoesNotExists')),
+  assert_false(is_data_property(test:'p')),
+  assert_false(is_data_property(test:'bIndiv')),
+  assert_true(is_data_property(test:'P')),
+  % Argument is unbound
+  assert_true(is_data_property(_)),
+  assert_true(once((is_data_property(A),test:'P'=A))).
+
+test('is_functional_property') :-
+  assert_true(is_functional_property(test:'p')),
+  assert_false(is_functional_property(test:'s')),
+  assert_false(is_functional_property(test:'r')),
+  assert_false(is_functional_property(test:'DoesNotExists')),
+  % Argument is unbound
+  assert_true(is_functional_property(_)),
+  assert_true(once((is_functional_property(A),test:'p'=A))).
+
+test('is_transitive_property') :-
+  assert_false(is_transitive_property(test:'p')),
+  assert_false(is_transitive_property(test:'s')),
+  assert_true(is_transitive_property(test:'r')),
+  assert_false(is_transitive_property(test:'DoesNotExists')),
+  % Argument is unbound
+  assert_true(is_transitive_property(_)),
+  assert_true(once((is_transitive_property(A), test:'r'=A))).
+
+test('is_symmetric_property') :-
+  assert_false(is_symmetric_property(test:'p')),
+  assert_true(is_symmetric_property(test:'s')),
+  assert_false(is_symmetric_property(test:'r')),
+  assert_false(is_symmetric_property(test:'DoesNotExists')),
+  % Argument is unbound
+  assert_true(is_symmetric_property(_)),
+  assert_true(once((is_symmetric_property(A), test:'s'=A))).
+
+test('has_inverse_property') :-
+  assert_true(has_inverse_property(test:'r',test:'rInv')),
+  assert_true(has_inverse_property(test:'rInv2',test:'r')),
+  % Left argument is unbound
+  assert_true(has_inverse_property(test:'r',_)),
+  % Right argument is unbound
+  assert_true(has_inverse_property(_,test:'rInv')),
+  % Both arguments are unbound
+  assert_true(has_inverse_property(_,_)),
+  % Negative Case
+  assert_false(has_inverse_property(test:'p',test:'r')).
+
+test('has_property_chain') :-
+  % Left argument is unbound
+  assert_true(
+    has_property_chain(_,[test:'t1',test:'t2',test:'t3'])
+  ),
+  assert_true(
+    once((
```

Comment:
we could define `assert_once` in rosprolog test interface

Reply:
Should this be resolved for this pullrequest, or should I create a ticket?

Reply:
no, we do not need to resolve it here. it was just an obersavation